src/environmentd/src/http/webhook.rs

```diff
@@ -129,12 +186,16 @@ pub enum WebhookError {
     Unsupported(&'static str),
     #[error("failed to deserialize body as {ty:?}: {msg}")]
     InvalidBody { ty: ScalarType, msg: String },
+    #[error("failed to validate the request: {0}")]
+    ValidationFailed(String),
+    #[error("attempt to validate request panicked")]
```

Comment:
This is really cool! -- a panic during evaluation is caught and does not impact the rest of the environmentd:
```
CREATE SOURCE s2 FROM WEBHOOK
  BODY FORMAT BYTES
  VALIDATE USING (
    mz_internal.mz_panic(headers::text) is null );
```

A test for that would be nice, as a future refactoring may preturb your panic handling.